lib/Algo2023Day04.cs

```diff
@@ -8,32 +10,112 @@ namespace Lib
-                        if(gamePoint == 0)
+                        if (winningNumbers.Contains(playNumber))
                         {
-                            gamePoint++;
+                            gamePoint *= 2;
+                            if(gamePoint == 0)
+                            {
+                                gamePoint++;
+                            }
                         }
                     }
+                    totalPoint += gamePoint;
+
                 }
-                totalPoint += gamePoint;
+                return totalPoint.ToString();
+            }
+            string bonus(string[] input)
+            {
+
+                List<Card> cards = new List<Card>();
+                foreach (var line in input)
+                {
+                    string[] cardGame = line.Split(':')[0].Split(' ');
+                    int cardId = int.Parse(cardGame.Where(e => e != "").ToArray()[1]);
+                    List<int> winningNumbers = line
+                        .Split(':')[1]
+                        .Split('|')[0]
+                        .Split(' ')
+                        .ToList()
+                        .Where(str => !string.IsNullOrWhiteSpace(str))
+                        .Select( str => int.Parse(str))
+                        .ToList();
+                    List<int> playNumbers = line
+                        .Split(':')[1]
+                        .Split('|')[1]
+                        .Split(' ')
+                        .ToList()
+                        .Where(str => !string.IsNullOrWhiteSpace(str))
+                        .Select(str => int.Parse(str))
+                        .ToList();
+                    Card card = new Card(cardId, winningNumbers, playNumbers);
+                    cards.Add(card);
+                }
+                for(int i = 1;i <= 202;i ++)
+                {
+                    // rechercher la carte dont la référence est 'i'
+                    List<Card> currenCardModel = [.. cards.Where( c => c.ReferenceId == i)];
+                    
+                    // TODO résoudre le nombre de carte gagné avec ce modèle de carte
 
+                    int winCount = 0;
+                    foreach(int playNumber in cards[i].Numbers)
+                    {
+                        if (cards[i].WinningNumbers.Contains(playNumber))
+                        {
+                            winCount++;
+                        }
+                    }
+
+
+                    // TODO trouver les modèles de cartes suivant (i + winCount) currenCardModel.Count foi
+                    for (int j = 0;j < currenCardModel.Count;j++)
+                    {
+                        for(int k = i+1; k < i+winCount+1; k++)
+                        {
+                            // ajouter les carte doublon jusqua i+wincount
+                            if (k < 202)
+                            {
+                                Card addCard = new Card(cards[k]);
```

Comment:
Attention, on ne cherche pas la cellule suivante du tableau, mais une carte dont la référence est "i+k"

```suggestion
Card originalWon = cardCollection.Find(x => x.ReferenceId == nextReferenceWon);
```